modules/transactions.js

```diff
@@ -311,97 +304,115 @@ __private.getPooledTransactions = function (method, req, cb) {
- * @return {function} Calls transactionPool.transactionInPool
+ * @return {boolean} False if transaction not in pool.
  */
 Transactions.prototype.transactionInPool = function (id) {
-	return __private.transactionPool.transactionInPool(id);
-};
-
-/**
- * @param {string} id
- * @return {function} Calls transactionPool.getUnconfirmedTransaction
- */
-Transactions.prototype.getUnconfirmedTransaction = function (id) {
-	return __private.transactionPool.getUnconfirmedTransaction(id);
+	var transaction = library.logic.transactionPool.get(id);
+	return transaction.status === 'Transaction not in pool' ? false : true;
 };
 
 /**
+ * Gets transaction from pool and check if is in pending signature list.
+ * @implements {transactionPool.get}
  * @param {string} id
- * @return {function} Calls transactionPool.getQueuedTransaction
- */
-Transactions.prototype.getQueuedTransaction = function (id) {
-	return __private.transactionPool.getQueuedTransaction(id);
-};
-
-/**
- * @param {string} id
- * @return {function} Calls transactionPool.getMultisignatureTransaction
+ * @return {transaction|undefined} Calls transactionPool.getMultisignatureTransaction
  */
 Transactions.prototype.getMultisignatureTransaction = function (id) {
-	return __private.transactionPool.getMultisignatureTransaction(id);
+	var transaction = library.logic.transactionPool.get(id);
+	return transaction.status === 'pending' ? transaction.transaction : undefined;
 };
 
 /**
- * Gets unconfirmed transactions based on limit and reverse option.
- * @param {boolean} reverse
+ * Gets unconfirmed transactions based on limit.
+ * @implements {transactionPool.getReady}
  * @param {number} limit
- * @return {function} Calls transactionPool.getUnconfirmedTransactionList
+ * @return {transaction[]}
  */
-Transactions.prototype.getUnconfirmedTransactionList = function (reverse, limit) {
-	return __private.transactionPool.getUnconfirmedTransactionList(reverse, limit);
+Transactions.prototype.getUnconfirmedTransactionList = function (limit) {
+	return library.logic.transactionPool.getReady(limit);
 };
 
 /**
  * Gets queued transactions based on limit and reverse option.
+ * @implements {transactionPool.getAll}
  * @param {boolean} reverse
  * @param {number} limit
- * @return {function} Calls transactionPool.getQueuedTransactionList
+ * @return {transaction[]}
  */
 Transactions.prototype.getQueuedTransactionList = function (reverse, limit) {
-	return __private.transactionPool.getQueuedTransactionList(reverse, limit);
+	return library.logic.transactionPool.getAll('unverified', {reverse: reverse, limit: limit});
 };
 
 /**
- * Gets multisignature transactions based on limit and reverse option.
+ * Gets multisignature transactions based on limit.
+ * @implements {transactionPool.getAll}
  * @param {boolean} reverse
  * @param {number} limit
  * @return {function} Calls transactionPool.getQueuedTransactionList
  */
 Transactions.prototype.getMultisignatureTransactionList = function (reverse, limit) {
```

Comment:
Do we need all those wrappers functions?

Reply:
Yes, we need to rewrite all internal calls, that is for integration phase